### PR TITLE
Example unit tests file + github actions

### PR DESCRIPTION
**Pull request by preeth1**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Python 3
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
```
Comment (jessicamillar) [inline .github/workflows/ci.yaml:18]: I'm on python 3.8.6, can we use that?

File: gw_spaceheat/test/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        # check that s.split fails when the separator is not a string
+        with self.assertRaises(TypeError):
+            s.split(2)
+
+    def test_failure(self):
+        self.assertEqual(1, 1000)
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (jessicamillar) [inline gw_spaceheat/test/test_examples.py:25]: I am guessing the red circle means the test failed?

**Review by jessicamillar — COMMENTED** [review]
Looks good, wondering about python version matching 3.8.6 of venv?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install nose
+      - name: Run tests with nosetest
+        run: nosetests gw_spaceheat/
```
[the remark this answers is not in the document]
Comment (preeth1) [reply .github/workflows/ci.yaml]: It means that there's no newline at the end of a file- just added one here:
https://github.com/thegridelectric/gw-scada-spaceheat-python/pull/4/commits/c7f8e9abf2366ac066d4cad195d13d8f73747b94
I'm not sure why github specifically shows it like it's a big error or something, when it's actually just a formatting thing- but this is the kind of thing that stickler-ci will be able to catch!

File: gw_spaceheat/test/test_examples.py (hunk 2 above)
Comment (preeth1) [reply gw_spaceheat/test/test_examples.py:25]: addressed what it is here: https://github.com/thegridelectric/gw-scada-spaceheat-python/pull/4#discussion_r879975851
All the info on the unit tests failing can be accessed here: https://github.com/thegridelectric/gw-scada-spaceheat-python/pull/4#discussion_r879975851
^^ I can give more of a tutorial on how github actions works during our next pairing session - a lot of it is visual stuff and knowing where to click (conducive to a screenshare)
Screen Shot 2022-05-23 at 9 19 53 PM

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (preeth1) [reply .github/workflows/ci.yaml]: Good call, updated it here: https://github.com/thegridelectric/gw-scada-spaceheat-python/pull/4/commits/8c61cb27f5ee0010dda6f28f03a42f418c56d317

**Review by jessicamillar — APPROVED** [review]
looks good - and I am ready to start adding unit tests :)

**Pull request merged**